Utils/CommonDriver.cs

```diff
@@ -0,0 +1,149 @@
+﻿using AventStack.ExtentReports;
+using AventStack.ExtentReports.Reporter;
+using CompetitionTask.Pages;
+using NUnit.Framework;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Chrome;
+using System;
+using System.IO;
+
+
+namespace CompetitionTask.Utils
+{
+    public class CommonDriver
+    {
+        CommonDriver commonDriverObj;
+        LoginPage loginPageObj;
+        Education educationObj;
+        Certificate certificateObj;
+
+        public static IWebDriver driver;
+        public static ExtentReports extent;
+        public static ExtentTest test;
+        public static List<string> DegreesToDelete { get; set; } = new List<string>();
+        public static List<string> CertificatenamesToDelete { get; set; } = new List<string>();
+
+
+
+
+        [OneTimeSetUp]
+        public void ExtentReportSetup()
+        {
+            try
+            {
+                
+                var sparkReporter = new ExtentSparkReporter(@"D:\Mansi-Industryconnect\CompetitionTask\Reports\extentReport.html");
+                extent = new ExtentReports();
+                extent.AttachReporter(sparkReporter);
+
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e.ToString());
+            }
+        }
+
+        [SetUp]
+        public void Initialization()
+        {
+            // Initialize WebDriver (Chrome in this case)
+
+            
+            driver = new ChromeDriver();
+            driver.Manage().Window.Maximize();
+            driver.Navigate().GoToUrl("http://localhost:5000/Home");
+            loginPageObj = new LoginPage();
+            educationObj = new Education();
+            certificateObj = new Certificate();
+
+            var testName = TestContext.CurrentContext.Test.Name;
+            test = extent.CreateTest(testName);
+
+            loginPageObj.LoginAction(driver);
+            test.Log(Status.Info, "Login successful");
+        }
+
+        [TearDown]
+        public void Cleanup()
```

Comment:
so your clean up is still not correct!

Reply:
the clean all data should have in OneTimeSetUp

and then for each test you should only clean up data that the test is generating in the teardown  but your cleanup method will be different as it will take information on what data needs to be cleaned -> cleanup(EducationDataToCleanUp)